src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -512,9 +512,45 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                 var response = new HttpResponseMessage() { RequestMessage = request, Content = new HttpConnectionResponseContent() };
                 ParseStatusLine(await ReadNextResponseHeaderLineAsync().ConfigureAwait(false), response);
 
-                // If we sent an Expect: 100-continue header, handle the response accordingly. Note that the developer
-                // may have added an Expect: 100-continue header even if there is no Content.
-                if (hasExpectContinueHeader)
+                // Multiple 1xx responses handling.
+                // RFC 7231: A client MUST be able to parse one or more 1xx responses received prior to a final response,
+                // even if the client does not expect one. A user agent MAY ignore unexpected 1xx responses.
+                // In .NET Core, apart from 100 Continue, and 101 Switching Protocols, we will treat all other 1xx responses
+                // as unknown, and will discard them.
+                while ((int)response.StatusCode >= 100 && (int)response.StatusCode <= 199)
```

Comment:
Consider instead:
```C#
while ((uint)(response.StatusCode - 100) <= 199 - 100)
```
as a minor optimization.

Reply:
I can see this optimization reduces comparison condition, but why using `uint`?

Reply:
> why using uint

Imagine that response.StatusCode is 20.  20 - 100 == -80, and if we were comparing by signed values, then -80 <= 199 - 100.  But by casting to uint, -80 wraps around to be 4294967216, which then fails the comparison.